File: e2e/playwright-typescript/tests/task-info/task-info.axe.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import { AllureEpic } from '@/data/AllureEpic';
import * as allure from "allure-js-commons";

test.beforeAll(async () => {
  await allure.epic(AllureEpic.ACCESSIBILITY);
  await allure.feature('View task info');
  await allure.tms('102');
});

test.describe('View task info - accessibility', () => {
  let context: TaskContext;

  test.beforeEach(async ({ support }) => {
    context = new TaskContext();
    const response = context.createTaskResponse();

    await support.mock.api.getTasks([response]);
    await support.mock.api.getTask(response.id, response);
    await support.mock.api.getIsValid(response.id, true);
  });

  test('should have no accessibility violations on task info modal', { tag: '@accessibility' }, async ({ step, validate }) => {
    // given
    await step.navigation.openMainPage();
    await step.tasks.openTaskInfoForm(context.title);

    // when
    const results = await step.accessibility.analyze();

    // then
    await validate.accessibility.hasNoViolations(results);
  });
});

File: e2e/capybara-ruby/spec/task_info/task_info_axe_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'

RSpec.describe 'View task info - accessibility', type: :system, accessibility: true,
                                                  epic: AllureEpic::ACCESSIBILITY,
                                                  feature: 'View task info',
                                                  tms: '102' do
  let(:context) { TaskContext.new }

  before do
    response = context.create_task_response
    support.mock.api
         .get_tasks([response])
         .get_task(response[:id], response)
         .get_is_valid(response[:id], true)
  end

  it 'should have no accessibility violations on task info modal' do
    # given
    step.navigation.open_main_page
    step.tasks.open_task_info_form(context.title)

    # when
    results = step.accessibility.analyze

    # then
    validate.accessibility.has_no_violations(results)
  end
end
